app/assets/javascripts/discourse/app/components/sidebar/messages-section.js

```diff
@@ -37,27 +44,38 @@ export default class SidebarMessagesSection extends GlimmerComponent {
     currentRouteParentName,
     currentRouteParams,
   }) {
-    const sectionLinks = [
-      ...this.personalMessagesSectionLinks,
-      ...this.groupMessagesSectionLinks,
-    ];
-
-    if (currentRouteParentName !== "userPrivateMessages") {
-      sectionLinks.forEach((sectionLink) => {
+    if (
+      currentRouteParentName !== "userPrivateMessages" &&
+      currentRouteParentName !== "topic"
+    ) {
+      for (const sectionLink of this.allSectionLinks) {
         sectionLink.collapse();
-      });
+      }
     } else {
-      sectionLinks.forEach((sectionLink) => {
-        sectionLink.pageChanged(currentRouteName, currentRouteParams);
-      });
+      const attrs = {
+        currentRouteName,
+        currentRouteParams,
+      };
+
+      if (currentRouteParentName === "topic") {
+        const topicController = getOwner(this).lookup("controller:topic");
+
+        if (topicController.model.isPrivateMessage) {
+          attrs.privateMessageTopic = topicController.model;
+        }
+      }
+
+      for (const sectionLink of this.allSectionLinks) {
+        sectionLink.pageChanged(attrs);
+      }
     }
   }
 
   @cached
```

Comment:
Oh nice I didn't know about this option, I guess it's like ruby memoizing?

Reply:
It is not exactly like memoizing in Ruby. Getters are recalculated each time you call the getter but ember provides a way to cache the getter if there is tracked value in the getter. https://api.emberjs.com/ember/release/modules/@glimmer%2Ftracking%2Fprimitives%2Fcache will probably explain it better than I can xD